components/features/dashboard/groups/GroupsList.tsx

```diff
@@ -0,0 +1,67 @@
+"use client";
+
+import axios from "@/lib/axiosInstance";
+import { getCookie } from "cookies-next";
+import { useQuery } from "@tanstack/react-query";
+import { GroupType } from "@/types/Group";
+import { Skeleton } from "@/components/ui/skeleton";
+import { Users } from "lucide-react";
+type Props = {
+  schoolId: string;
+};
+
+async function getGroup(schoolId: string) {
+  const token = await getCookie("token");
+  const res = await axios.get(`/school/${schoolId}/group`, {
+    headers: { Authorization: token },
+  });
+  return res.data;
+}
+
+export default function GroupList({ schoolId }: Props) {
+  const { data: group, isLoading } = useQuery<GroupType[]>({
+    queryKey: ["groups"],
+    queryFn: () => getGroup(schoolId),
+  });
+
+  if (isLoading) {
+    return (
+      <div className="grid w-full grid-cols-1 gap-6 md:grid-cols-2 lg:grid-cols-3">
+        {[...Array(6)].map((_, index) => (
```

Comment:
A better approach, using index on key is not recommended.
```suggestion
        {[...Array(6).keys()].map((key) => (
```